### PR TITLE
[pos_disable_payment] Cleanup & Improvments

### LINKED ISSUE #508: [pos_disable_payment] PROPORSAL replicate behavior on pos.config
I'm planning to do a PR with the following characteristics:

- Cleanups (various)
- Replicate the very same limitation this module implements onto the pos.config model

This use case is warranted for a so called "order point" (spanish: "toma pedidos") in restaurants in the case where you need special authorizations on a device basis in order to "close orders", that is to emit the invoice. *example, you need a special authorized sequence or similar, which for synchronization reasons you need to assign to specific tablets rather than specific users...)

Would you be fine with this?
Any acceptance criteria to look for?

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/52104644-pos_disable_payment-proporsal-replicate-behavior-on-pos-config?utm_campaign=plugin&utm_content=tracker%2F2289114&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F2289114&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by blaggacao**
Closes #508 

@GabbasovDinar I guess you are the expert, here...

--------------------

Halleluya #505 !!! ... my odysee:

##### In local clone `../it-projects-llc/pos-addons `: 
- `git subrepo init pos_disable_payment`
##### In my working tree: 
1. `git subrepo clone --branch=subrepo/pos_disable_payment ../it-projects-llc/pos-addons it-projects-llc/pos_disable_payment`
2.  Do stuff (the changesets on `pos_disable_payment` you can see)
3. `git subrepo push it-projects-llc/pos_disable_payment`
##### In local clone of `../it-projects-llc/pos-addons `: 
- `git-subhistory merge -a pos_disable_payment subrepo/pos_disable_payment`

`git subrepo pull` can be tricked into this last step (`remote = .`) but does squash commit history, and I wanted you to be able to cherry pick.

### CODE REVIEW

**Comment by blaggacao** [comment]
Here is the changelog:
```
Changes since upstream v10:

  pos_disable_payment:
   - [CLEANUP] cleanup code and small refactorings
   - [REF] respect framework recommendations
   - [l18n] fix ambigous naming (+de, +es)
   - [REF] streamline code
   - [REF] align module structrue with official recomendation
   - [IMP] add restrictions on pos config
   - [IMP] imrpove UI of disabled buttons

  Contributors:
   - David Arnold

```

**Comment by yelizariev** [comment]
Could you please don't rename files. While you are right about naming, it leads to problems with merging updates from 9.0 to 10.0. So, please keep old names for existing files.

Second question. Are you sure, that ``.gitrepo`` files are needed in the repo? Does it help to other developers who use similar approach to development/deployment?

File: pos_disable_payment/static/src/js/pos_disable_payment.js [hunk 1]
```diff
@@ -9,190 +9,127 @@ odoo.define('pos_disable_payment', function(require){
...
-                    return el.id === users[i].id;
-                });
-                if (user) {
-                    _.extend(user,users[i]);
-                }
-            }
-        }
-    });
+    models.load_fields("res.users",  [      'allow_payments',      'allow_delete_order',      'allow_discount',      'allow_edit_price',      'allow_decrease_amount',      'allow_delete_order_line',      'allow_create_order_line',      'allow_refund'])
+    models.load_fields("pos.config", ['iface_allow_payments','iface_allow_delete_order','iface_allow_discount','iface_allow_edit_price','iface_allow_decrease_amount','iface_allow_delete_order_line','iface_allow_create_order_line','iface_allow_refund'])
```
Comment (GabbasovDinar) [inline pos_disable_payment/static/src/js/pos_disable_payment.js]: all fields in pos.config are loaded automatically, not necessary to use this construction 
https://github.com/odoo/odoo/blob/10.0/addons/point_of_sale/static/src/js/models.js#L227

Comment (blaggacao) [reply pos_disable_payment/static/src/js/pos_disable_payment.js]: Thank you, I've overlooked that!

**Comment by blaggacao** [comment]
@yelizariev 
- **Renaming:** Good point, haven't thought about that... I just wanted to make me feel "at ease" to be more efficient...
- **Gitrepo:** You are right, just as they are (without having a public bridge-repo) does not really help anyone. I was happy enough having been able to extract my changes for that moment. In a cleanup, I would remove the constituting subrepo commits. (this is where `git subrepo push` would be nothing short of perfect, if only the upstream repo structure would support it...)

I think I'll do a cleanup next week or so. For now I'm on a 90 min deadline pressure... :wink: 
But they'll feed me like a king, at least, if everything works.. :wine_glass:

**Comment by blaggacao** [comment]
FYI: I'll also push a pos_disable_payment_restaurant which takes care of the split-bill case as a matter of reciprocity and saying thanks to you guys...

**Pull request closed without merging**